File: 백준/Silver/1629. 곱셈/곱셈.py
def cal(a, b, c):
    if b <= 10:
        return (a ** b) % c
    else:
        return ((cal(a % c, b // 2, c) ** 2) * cal(a % c, b % 2, c)) % c

a, b, c = map(int, input().split())

print(cal(a % c, b, c))

File: 백준/Silver/1629. 곱셈/곱셈.cc
#include <iostream>

using namespace std;

typedef long long ll;

ll Cal(const ll& a, ll b, const ll& c) 
{
    if (b == 1) {
        return a % c;
    }
    
    ll num = Cal(a, b / 2, c) % c;
    if (b % 2 == 0) {
        return (num * num) % c;
    }
    else {
        return ((a % c) * ((num * num) % c)) % c;
    }
}

int main()
{
    ll a, b, c;
    cin >> a >> b >> c;
    
    cout << Cal(a, b, c) << '\n';
    return 0;
}